### DISCUSSION TITLE
Upstream doesn't match the same default branch as current origin

### DISCUSSION THREAD

**Discussion by sheldonhull** [git-town/git-town#3231] [General]
Was hoping to add upstream and then be able to run `git town hack` or `git sync` and essentially have it rebase or merge the upstream changes into the branch as well as origin main.

However, the upstream doesn't have the same branch of `main`. Instead upstream has `develop` and origin has `main`. 

Is there a way to configure this so the sync command works correctly? I've tried various ways with git config and branching commands, but git town seems to assume the upstream name is main and I can't get it to *not* do this.

**Comment by kevgo** [comment]
Good question! Git Town currently assumes that all branches have the same name in all remotes.

That's because Git requires that tracking branches have the same name as local branches: The tracking branch of `main` must be named `origin/main`. Because of YAGNI Git Town applies this rule for pulling updates of the main development branch from `upstream` for now.

If there is a good use case, we can make this more flexible. Does your `upstream/develop` branch contain the same commits as `origin/main`? If yes, can you please elaborate why your environment was set up this way, i.e. what are the advantages of this setup?

Also, just to be sure, can you please post the Git commands you run when you manually pull updates from `upstream/develop` into your local `main` branch, which you want to automate through Git Town? Thanks!
